File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/util/ConnectionInitiationHelper.java
package lunalauf.rms.utilities.network.util;

import java.util.Random;

public class ConnectionInitiationHelper {

	private static final Random random = new Random();

	public static String getSynMessage() {
		return String.valueOf(random.nextLong());
	}

	public static String getAckMessage(String synMessage) throws NumberFormatException {
		long number = Long.parseLong(synMessage);
		return String.valueOf(Math.abs(number) - 11);
	}

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/util/ConnectionInitiationHelper.kt
package lunalauf.rms.utilities.network.util

import java.util.*
import kotlin.math.abs

object ConnectionInitiationHelper {
    private val random = Random()
    val synMessage: String
        get() = random.nextLong().toString()

    @Throws(NumberFormatException::class)
    fun getAckMessage(synMessage: String): String {
        val number = synMessage.toLong()
        return (abs(number) - 11).toString()
    }
}
